### PR TITLE
Added Sphinx Documentation

### PR DESCRIPTION
**Pull request by bcl**
Run 'make docs' to update the apidocs.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.. src documentation master file, created by
+   sphinx-quickstart on Wed Apr  8 13:46:00 2015.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to src's documentation!
```
Comment (vpodzime) [inline docs/index.rst]: Do we really want this in the index?

**Comment by vpodzime** [comment]
Other than the question/comment above this looks good to me.

File: docs/index.rst (hunk 1 above)
Comment (bcl) [inline docs/index.rst]: oops, missed a spot. that's what happens when you don't give it a proper name when you run the first pass.

**Comment by bcl** [comment]
pushed

**Pull request closed without merging**